pydantic/_internal/_model_construction.py

```diff
@@ -231,14 +231,6 @@ def __getattr__(self, item: str) -> Any:
             private_attributes = self.__dict__.get('__private_attributes__')
             if private_attributes and item in private_attributes:
                 return private_attributes[item]
-            if item == '__pydantic_core_schema__':
-                # This means the class didn't get a schema generated for it, likely because there was an undefined reference
-                maybe_mock_validator = getattr(self, '__pydantic_validator__', None)
-                if isinstance(maybe_mock_validator, MockValSer):
-                    rebuilt_validator = maybe_mock_validator.rebuild()
-                    if rebuilt_validator is not None:
-                        # In this case, a validator was built, and so `__pydantic_core_schema__` should now be set
-                        return getattr(self, '__pydantic_core_schema__')
```

Comment:
Can this really be removed entirely? Maybe I'm missing something about how the logic you've added removes this requirement...

Reply:
I believe this is because of line 131 in `set_model_mocks` from `_mock_val_ser.py`, which sets it unconditionally, so you should never hit this now

Reply:
Well, I guess it comes down to whether there is any way that the MockValSer can be set _without_ using the defer_build stuff. If it can be set without the defer build then yeah it seems suspicious to delete this.

Reply:
Ah, it seems to be because of line 148 of main.py